addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -504,43 +488,52 @@ options.registry.WebsiteFormEditor = FormEditor.extend({
             if (fields.length) {
                 // ideally we'd only do this if saving the form
                 // succeeds... but no idea how to do that
-                this.orm.call("ir.model.fields", "formbuilder_whitelist", [model, unique(fields)]);
+                this.env.services.orm.call("ir.model.fields", "formbuilder_whitelist", [model, unique(fields)]);
             }
         }
         if (this.$message.length) {
             this.$target.removeClass('d-none');
             this.$message.addClass("d-none");
         }
-    },
+    }
     /**
      * @override
      */
-    updateUI: async function () {
-        // If we want to rerender the xml we need to avoid the updateUI
-        // as they are asynchronous and the ui might try to update while
-        // we are building the UserValueWidgets.
-        if (this.rerender) {
-            this.rerender = false;
-            await this._rerenderXML();
-            return;
-        }
-        await this._super.apply(this, arguments);
+    async updateUI() {
+        await super.updateUI(...arguments);
         // End Message UI
         this.updateUIEndMessage();
-    },
+        // TODO: @owl-options was in renderCustomXML
+        const formKey = this.activeForm.website_form_key;
+        const formInfo = FormEditorRegistry.get(formKey, null);
+        formInfo.fields.forEach(field => {
+            if (field.required) {
+                // Try to retrieve hidden value in form, else,
+                // get default value or for many2one fields the first option.
+                const currentValue = this.$target.find(`.s_website_form_dnone input[name="${field.name}"]`).val();
+                const defaultValue = field.defaultValue || field.records[0].id;
+                // TODO this code is not rightfully placed (even maybe
+                // from the original form feature in older versions). It
+                // changes the $target while this method is only about
+                // declaring the option UI. This for example forces the
+                // 'email_to' value to a dummy value on contact us form just
+                // by clicking on it.
+                this._addHiddenField(currentValue || defaultValue, field.name);
+            }
+        });
+    }
```

Comment:
Would this be fixable before I cherry-pick the commit? If not, can you add a comment as to why we aren't changing it in this commit?